### PR TITLE
feat: bump dependencies to support Kubernetes 1.23

### PR DESCRIPTION
**Pull request by chrisgacsal**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| API breaks?     | yes
| Deprecations?   | no
| Related tickets | #92 
| License         | Apache 2.0

### What's in this PR?
Update dependencies to support Kubernetes 1.23. 

This also introduces a breaking change related to `logging.logger` which can no longer be used as implementation of `logr.Logger` interface due to the changes introduced in `go-logr/logr` v1.0. In order to keep the original functionality of the `logging` package, the `logr.Logger` is replaced with custom `logging.Logger` interface.

### Checklist

- [x] Code meets the [Developer Guide](https://github.com/banzaicloud/developer-guide)
- [x] User guide and development docs updated (if needed)
- [x] Related Helm chart(s) updated (if needed)

### CODE REVIEW

**Review by baluchicken — APPROVED** [review]
LGTM thx @chrisgacsal

**Review by pepov — APPROVED** [review]

**Review by baluchicken — APPROVED** [review]

File: pkg/logger/logsink.go [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+		log.spinner.FinalMSG = color.New(log.colors.Info).Sprintf("%c", log.checkMark) + log.spinner.Suffix + "\n"
+	}
+	log.mux.Unlock()
+
+	if log.spinner != nil && !log.grouped {
+		log.stopSpinner()
+	}
+}
+
+// Enabled  implements logr.LogSink interface
```
Comment (hi-im-aren) [inline pkg/logger/logsink.go:126]:
```suggestion
// Enabled implements logr.LogSink interface
```

**Review by hi-im-aren — APPROVED** [review]

**Review by waynz0r — APPROVED** [review]

File: pkg/logger/logsink.go [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+	l := log.copyLogger()
+	l.names = append(l.names, name)
+
+	return l
+}
+
+// WithValues implements logr.LogSink interface
+func (log *spinnerLogSink) WithValues(keysAndValues ...interface{}) logr.LogSink {
+	l := log.copyLogger()
+	l.values = append(l.values, keysAndValues)
```
Comment (pepov) [inline pkg/logger/logsink.go:176]:
```suggestion
	l.values = append(l.values, keysAndValues...)
```

**Review by pepov — CHANGES_REQUESTED** [review]

File: pkg/logger/logsink.go (hunk 2 above)
Comment (chrisgacsal) [reply pkg/logger/logsink.go]: Nice catch! Ty

**Pull request merged**